test/lib/deployment/test-deployment.js

```diff
@@ -178,7 +178,29 @@ async function runProbe(probe, deploymentId, deploymentUrl, ctx) {
     fetchOpts.headers['content-type'] = 'application/json';
     fetchOpts.body = JSON.stringify(probe.body);
   }
-  const { text, resp } = await fetchDeploymentUrl(probeUrl, fetchOpts);
+  let result = await fetchDeploymentUrl(probeUrl, fetchOpts);
+
+  // If we receive the preview page from Vercel, the real page should appear momentarily,
+  // retry a few times before running the probe checks
+  const checkForPreviewPage = text => {
+    return text.includes('This page will update once the build completes');
+  };
+  let isShowingBuildPreviewPage = checkForPreviewPage(result.text);
+  for (let retryCount = 0; retryCount < 10; retryCount++) {
+    if (!isShowingBuildPreviewPage) {
+      break;
+    } else {
+      result = await fetchDeploymentUrl(probeUrl, fetchOpts);
+      isShowingBuildPreviewPage = checkForPreviewPage(result.text);
```

Comment:
<!-- Suggestion comments propose improvements to the code. Mention specifically what should be changed and what the benefit would be.  -->
**suggestion (non-blocking):** break early to avoid extra 1s delay‏

```suggestion
      isShowingBuildPreviewPage = checkForPreviewPage(result.text);
      if (!isShowingBuildPreviewPage) {
        break;
      }
```